pkg/operator/k8sutil/network.go

```diff
@@ -110,3 +112,38 @@ func ParseLinuxIpAddrOutput(rawOutput string) ([]LinuxIpAddrResult, error) {
 	}
 	return results, nil
 }
+
+func GetIpAddressType(addresses []string) (discoveryv1.AddressType, error) {
+	var addressType discoveryv1.AddressType
+
+	if len(addresses) == 0 {
+		return addressType, errors.Errorf("failed to get the address type for ip address, address list is empty type")
+	}
+	ip, err := netip.ParseAddr(addresses[0])
+	if err != nil {
+		return addressType, errors.Errorf("failed to get the address type for ip address %q", addresses[0])
+	}
+	addressType = discoveryv1.AddressTypeIPv4
+	if ip.Is6() {
+		addressType = discoveryv1.AddressTypeIPv6
+	}
+
+	for _, address := range addresses {
```

Comment:
```suggestion
for _, addr := range addresses[1:] {
```

for better readability  and no need to parse the first IP again?